### DISCUSSION TITLE
invocation tree的具体作用是什么

### DISCUSSION THREAD

**Discussion by A491382** [alibaba/Sentinel#2099] [Q&A]
entranceNode和defaultnode会构建具体的invocation tree ， 但是比如一个resource可能被不同的调用关系调用 ， 比如A->B->C , 构建了一个invocation tree ， 但是同一个contextname也可能出现 B->D ，但是这个invocation tree就不能从entranceNode开始生成 ，相当于并不是所有的调用关系都能从entranceNode开始，那么记录这个invocation tree在进行汇总统计的时候，比如统计这个调用链A->B->C  ， 会误把B->D里面的计数也统计进去呢 ？还是说出现这种问题要通过不同的contextName把调用关系分离开，谢谢

**Comment by zidoshare** [comment]
所有的调用都是entranceNode开始的。

代码如下：
![Screenshot from 2021-04-02 15-56-49](https://user-images.githubusercontent.com/11310504/113395340-42e75380-93cc-11eb-9c57-35ee4b3b5fa1.png)

查看一下invacation tree
![Screenshot from 2021-04-02 15-57-08](https://user-images.githubusercontent.com/11310504/113395343-44b11700-93cc-11eb-839a-eeb5d9fb3566.png)

**Reply by A491382** [reply]
是的， 但是不是很理解这个invocationtree的意义， 假如没有这个tree会怎么样呢， 因为nodeselectorslot里面只是使用contextname决定是否生成node

**Reply by SomebodyHuang2017** [reply]
单机限流还支持两种比较高级的策略：1. 关联限流 2. 链路限流
有了流控资源调用链路关系，就可以支持这两种策略。

具体的案例可以看阿里云高可用文档：https://help.aliyun.com/document_detail/101077.html
